Common/tests/ResultWatcherTestBase.cs

```diff
@@ -0,0 +1,326 @@
+                                       "OwnerId2",
+                                       ResultStatus.Created,
+                                       DateTime.Today,
+                                       new[]
+                                       {
+                                         (byte)1,
+                                       }),
+                            new Result("SessionId2",
+                                       "ResultIsCreated3",
+                                       "OwnerId3",
+                                       ResultStatus.Created,
+                                       DateTime.Today,
+                                       new[]
+                                       {
+                                         (byte)1,
+                                       }),
+                          })
+                  .Wait();
+    }
+  }
+
+  [TearDown]
+  public virtual void TearDown()
+  {
+    ResultTable   = null;
+    ResultWatcher = null;
+    RunTests      = false;
+  }
+
+  /* Interface to test */
+  protected IResultTable?   ResultTable;
+  protected IResultWatcher? ResultWatcher;
+
+  /* Boolean to control that tests are executed in
+   * an instance of this class */
+  protected bool RunTests;
+
+  /* Function be override so it returns the suitable instance
+   * of ResultTable and ResultWatcher to the corresponding interface implementation */
+  public virtual void GetInstance()
+  {
+  }
+
+  [Test]
+  public async Task InitShouldSucceed()
+  {
+    if (RunTests)
+    {
+      Assert.AreNotEqual(HealthStatus.Healthy,
+                         (await ResultWatcher!.Check(HealthCheckTag.Liveness)
+                                              .ConfigureAwait(false)).Status);
+      Assert.AreNotEqual(HealthStatus.Healthy,
+                         (await ResultWatcher.Check(HealthCheckTag.Readiness)
+                                             .ConfigureAwait(false)).Status);
+      Assert.AreNotEqual(HealthStatus.Healthy,
+                         (await ResultWatcher.Check(HealthCheckTag.Startup)
+                                             .ConfigureAwait(false)).Status);
+
+      await ResultWatcher.Init(CancellationToken.None)
+                         .ConfigureAwait(false);
+
+      Assert.AreEqual(HealthStatus.Healthy,
+                      (await ResultWatcher.Check(HealthCheckTag.Liveness)
+                                          .ConfigureAwait(false)).Status);
+      Assert.AreEqual(HealthStatus.Healthy,
+                      (await ResultWatcher.Check(HealthCheckTag.Readiness)
+                                          .ConfigureAwait(false)).Status);
+      Assert.AreEqual(HealthStatus.Healthy,
+                      (await ResultWatcher.Check(HealthCheckTag.Startup)
+                                          .ConfigureAwait(false)).Status);
+    }
+  }
+
+
+  private static async Task ProduceEvents(IResultTable      resultTable,
```

Comment:
It will be really useful to have some text specifying which events this function generate, and then at each test, how the events are updated and the expected behavior of the test (even if this is implicitly stated in the test name)